.git_hooks/pre-push

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+if ! cargo test ; then
```

Comment:
One more thing I thought of: this is a good opportunity to fix the intermittent 'too many open files' errors:

```
thread 'r2d2-worker-2' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 24, kind: Other, message: "Too many open files" }', /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/postgres-0.17.5/src/config.rs:318:27
```
```suggestion
ulimit -n 4096
if ! cargo test ; then
```

Reply:
To prevent trying to set 4096 and it being over the hard limit I inserted a check before it.
Also keep in mind that these settings won't persist in your env but only during each pre-push script running.
Furthermore, `ulimit -n` isn't POSIX compatible but I think it's safe enough, I literally couldn't find access to an old enough Linux that won't have `ulimit -n` working on it 😄 

https://github.com/koalaman/shellcheck/wiki/SC2039

> In .git_hooks/pre-push line 6:
>  ulimit -n "$fd_hard_limit"
>        ^-- SC2039: In POSIX sh, ulimit -n is undefined.

Reply:
I'm not worried about compatibility with very old Linux machines; if you're using something that old you'll know what the equivalent of `ulimit -n` is. The larger issue is that `cargo run build crate` only works on Linux currently (I've been meaning to open an issue).